### PR TITLE
Rx Completable for ApolloPrefetch

### LINKED ISSUE #389: RxApollo doesn't have a wrapper function for ApolloPrefetch

### PR DESCRIPTION
**Pull request by VisheshVadhera**
closes #389

### CODE REVIEW

File: apollo-rxsupport/src/main/java/com.apollographql.android.rx/RxApollo.java [hunk 1]
```diff
@@ -103,6 +106,37 @@ private RxApollo() {
...
+   */
+  @Nonnull public static Completable from(@Nonnull final ApolloPrefetch prefetch) {
+    checkNotNull(prefetch, "prefetch == null");
+
+    return Completable.create(new Completable.OnSubscribe() {
+      @Override public void call(CompletableSubscriber subscriber) {
+        cancelOnCompletableUnsubscribe(subscriber, prefetch);
+        try {
+          prefetch.execute();
+          subscriber.onCompleted();
```
Comment (sav007) [inline apollo-rxsupport/src/main/java/com.apollographql.android.rx/RxApollo.java]: Do we need to check:
```
if (!subscriber.isUnsubscribed()) {
}
```

File: apollo-rxsupport/src/main/java/com.apollographql.android.rx/RxApollo.java [hunk 2]
```diff
@@ -103,6 +106,37 @@ private RxApollo() {
...
+        } catch (ApolloException e) {
+          Exceptions.throwIfFatal(e);
+          subscriber.onError(e);
```
Comment (sav007) [inline apollo-rxsupport/src/main/java/com.apollographql.android.rx/RxApollo.java]: Do we need to check:
```
if (!subscriber.isUnsubscribed()) {
}
```

Comment (VisheshVadhera) [inline apollo-rxsupport/src/main/java/com.apollographql.android.rx/RxApollo.java]: Done

File: apollo-rxsupport/src/main/java/com.apollographql.android.rx/RxApollo.java (hunk 1 above)
Comment (VisheshVadhera) [inline apollo-rxsupport/src/main/java/com.apollographql.android.rx/RxApollo.java]: Done

File: apollo-integration/src/test/java/com/apollographql/apollo/RxApolloTest.java [hunk 3]
```diff
@@ -88,7 +92,7 @@
     TestSubscriber<EpisodeHeroName.Data> subscriber = new TestSubscriber<>();
     Subscription subscription = RxApollo
         .from(call)
-        .delay(5, TimeUnit.SECONDS)
+        .delay(RX_DELAY_SECONDS, TimeUnit.SECONDS)
```
Comment (digitalbuddha) [inline apollo-integration/src/test/java/com/apollographql/apollo/RxApolloTest.java]: what is point of the delay? will this slow our tests down by 5 seconds?

Comment (VisheshVadhera) [inline apollo-integration/src/test/java/com/apollographql/apollo/RxApolloTest.java]: The docs for delay say:

> This version of delay operates by default on the computation Scheduler.

which means that its a non blocking call and hence wont block the thread running the test. So the reason behind introducing delay is to let the observable do work in another thread while in the main thread we try to unsubscribe from the it. Basically simulate a real world scenario of unsubscribing from a Subscription.

File: apollo-integration/src/test/java/com/apollographql/apollo/RxApolloTest.java [hunk 4]
```diff
@@ -29,13 +29,17 @@
...
 import rx.observers.TestSubscriber;
 
 import static com.google.common.truth.Truth.assertThat;
 
 public class RxApolloTest {
+
   private ApolloClient apolloClient;
   private MockWebServer server;
+
+  private static final int RX_DELAY_SECONDS = 5;
```
Comment (sav007) [inline apollo-integration/src/test/java/com/apollographql/apollo/RxApolloTest.java]: we can reduce it to 2 I guess

Comment (VisheshVadhera) [inline apollo-integration/src/test/java/com/apollographql/apollo/RxApolloTest.java]: Done

**Pull request merged**